aten/src/ATen/native/native_functions.yaml

```diff
@@ -5287,19 +5287,51 @@
     CUDA: nested_from_padded_cuda
   autogen: _nested_from_padded.out
 
+# These private functions are temporary. They will be updated/deleted when nested tensors switch to using SymInts for their metadata representation
 - func: _nested_tensor_size(Tensor self) -> Tensor
   variants: method
   dispatch:
     NestedTensorCPU, NestedTensorCUDA: _nested_tensor_size
   autogen: _nested_tensor_size.out
 
+- func: _nested_tensor_strides(Tensor self) -> Tensor
+  variants: method
+  dispatch:
+    NestedTensorCPU, NestedTensorCUDA: _nested_tensor_strides
+  autogen: _nested_tensor_strides.out
+
+- func: _nested_tensor_offsets(Tensor self) -> int[]
+  variants: method
+  dispatch:
+    NestedTensorCPU, NestedTensorCUDA: _nested_tensor_offsets
+
 # _nested_from_padded is not usable from Python, so
 # _nested_from_padded_and_nested_example is available for testing.
 - func: _nested_from_padded_and_nested_example(Tensor padded, Tensor nt_example) -> Tensor
   dispatch:
     NestedTensorCPU, NestedTensorCUDA: NestedTensor_from_padded_and_nested_example
   autogen: _nested_from_padded_and_nested_example.out
 
+# The input arguments' types to this functions are temporary. When nested tensors switch to using SymInts for their metadata representation
+# this will need to be updated
+- func: _nested_from_buffer(Tensor(a) self, Tensor nested_size, Tensor nested_strides, int[] offsets) -> Tensor(a)
```

Comment:
How does this one differ from _nested_view_from_buffer_copy and why doesn't it have a restricted dispatch key set?

Reply:
Why does it also need offsets? I thought offsets are just a particular view of `nested_strides` used as an input  to functions that need it. Does it actually need to be provided or can it be made optional as a performance improvement?

Reply:
This is spiritual the same as reshape, I.e. a maybe view function. All maybe view functions need to be composite. It is maybe view because of the call to contiguous which is required to handle broadcasting right now. I had a version where strides and offsets are optional but since for now this is only ever callable by autorgrad in c++ I thought it was fine to keep it this way.

Reply:
Ok fine, then let's leave it in.